### ISSUE TITLE
Allow users to revoke an invite they sent

### ISSUE THREAD

**Issue by dkasak** [matrix-org/synapse#9582]
**Description:**

A user should be allowed to revoke an invite they sent even if they do not have the necessary power level to be able to *kick* the user after they join. This is useful in cases where you mistakenly invite the user into the wrong room. It is also the most intuitive behaviour from a user's perspective, IMO.

Conceptually, if we view this as a state diagram composed of states (as nodes) and actions (as arrows), any `* -> leave` arrow where the `sender` and `state_key` do not match is currently considered a "kick".

I propose that we allow an additional `invite -> leave` "disinvite" arrow which is allowed if the `sender` of the `invite` event is the same as the `sender` of a subsequent `leave` event.

**Comment by clokep** [comment]
Would this require a spec change?

**Comment by richvdh** [comment]
yes it would: https://github.com/matrix-org/matrix-doc/issues/664
